### DISCUSSION TITLE
Rendering/Formatting of Matplotlib plots within Trame

### DISCUSSION THREAD

**Discussion by EZoni** [Kitware/trame#579] [Q&A] [answered]
Hi all,

I'm new to Trame and I'm experimenting with rendering Matplotlib plots within Trame.

I noticed that I don't get the rendering that I would expect when it comes to axis labels formatting and such.

Here's a simple example. 

I start from your Matplotlib example at https://github.com/Kitware/trame/blob/master/examples/05_charts/Matplotlib/00_matplotlib-charts.py and scale the `x` array in the `Multilines` plot function by a very small number, say `1e-14` by relacing this line
https://github.com/Kitware/trame/blob/5f60a81a5f75f7f577ab2ed42e33c5b55256fa80/examples/05_charts/Matplotlib/00_matplotlib-charts.py#L66
with
```
    x = np.linspace(0, 10, 1000)*1e-14
```
which could be typical depending, for example, on the physical units behind those values.

If I launch the application and visualize the `Multilines` plot with the code change above, I get this plot:
<p align="center">
    <img src="https://github.com/user-attachments/assets/001acb45-2773-47e5-bb2f-36ccd0b5158e" width="700">
</p>

You can see that the x axis labels are not formatted automatically as they are if you run the plot function by itself, outside of Trame, which instead gives this plot:
<p align="center">
    <img src="https://github.com/user-attachments/assets/32201b43-7563-4a42-a16e-a34beb6d0e34" width="700">
</p>

How can I get Trame to format the axis labels automatically as Matplotlib does outside of Trame?

Note that manual formatting (e.g., `ax.xaxis.set_major_formatter(ticker.FormatStrFormatter('%0.0e'))`) does not seem to be effective either.

Is this maybe a limitation of the library Trame relies on to map the matplotlib definition from the server to the client, similarly to what is mentioned in another discusssion at https://github.com/Kitware/trame/discussions/317#discussioncomment-6889968?

Thanks in advance for your help!

**Comment by jourdain** [comment, marked as answer]
Indeed, I'm worried that is coming from a limitation of the [mpld3](https://github.com/mpld3/mpld3). Trame is currently relying on v0.5.7 (November 2021) which we could bump to v0.5.10 (December 2023) and see if that fix it. 

In general I would suggest to use plotly instead but I understand if you have experience with matplotlib it might be easier to stick with it.
